shared/lib/network-utils.ts

```diff
@@ -24,3 +55,62 @@ export async function getSafeChainsListFromCacheOnly(): Promise<ChainInfo[]> {
     return [];
   }
 }
+
+/**
+ * Determines whether the given RPC endpoint URL matches an Infura URL that uses
+ * our API key.
+ *
+ * @param endpointUrl - The URL of the RPC endpoint.
+ * @param infuraProjectId - Our Infura project ID.
+ * @returns True if the URL is an Infura URL, false otherwise.
+ */
+export function getIsMetaMaskInfuraEndpointUrl(
+  endpointUrl: string,
+  infuraProjectId: string,
+): boolean {
+  return new RegExp(
+    `^https://[^.]+\\.infura\\.io/v3/(?:\\{infuraProjectId\\}|${escapeRegExp(infuraProjectId)})$`,
```

Comment:
Note the addition of `{infuraProjectId}` to this regex. This change is necessary because the endpoint that is passed to this function has two variations. Either the URL comes from a network configuration, in which case the Infura API key is hidden (`{infuraProjectId}`) or it comes from `RpcService`, which has the full URL without hiding any API keys.